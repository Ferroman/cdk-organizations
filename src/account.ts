import { CustomResource } from "aws-cdk-lib";
import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from "aws-cdk-lib/custom-resources";
import { Construct, IConstruct } from "constructs";
import { AccountProvider } from "./account-provider";
import { IParent } from "./parent";
import { IPolicyAttachmentTarget } from "./policy-attachment";
/**
 * @see https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/control-access-billing.html#ControllingAccessWebsite-Activate
 */
export enum IamUserAccessToBilling {
  /**
   * If set to ALLOW, the new account enables IAM users to access account billing information if they have the required permissions.
   */
  ALLOW = "ALLOW",
  /**
   * If set to DENY, only the root user of the new account can access account billing information.
   */
  DENY = "DENY",
}

export interface AccountProps {
  /**
   * The email address of the owner to assign to the new member account. This email address must not already be associated with another AWS account. You must use a valid email address to complete account creation. You can't access the root user of the account or remove an account that was created with an invalid email address.
   */
  readonly email: string;
  /**
   * The friendly name of the member account.
   */
  readonly accountName: string;
  /**
   * The name of an IAM role that AWS Organizations automatically preconfigures in the new member account. This role trusts the management account, allowing users in the management account to assume the role, as permitted by the management account administrator. The role has administrator permissions in the new member account.
   *
   * If you don't specify this parameter, the role name defaults to OrganizationAccountAccessRole.
   */
  readonly roleName?: string;
  /**
   * If set to ALLOW , the new account enables IAM users to access account billing information if they have the required permissions. If set to DENY , only the root user of the new account can access account billing information.
   *
   * @default ALLOW
   */
  readonly iamUserAccessToBilling?: IamUserAccessToBilling;

  /**
   * The parent root or OU that you want to create the new Account in.
   */
  readonly parent?: IParent;
}

export interface IAccount extends IConstruct {
  /**
   * If the account was created successfully, the unique identifier (ID) of the new account. Exactly 12 digits.
   */
  readonly accountId: string;
  /**
   * The friendly name of the account.
   */
  readonly accountName: string;
  /**
   * The email address of the owner to assign to the new member account. This email address must not already be associated with another AWS account. You must use a valid email address to complete account creation. You can't access the root user of the account or remove an account that was created with an invalid email address.
   */
  readonly email: string;
}

/**
 * Creates an AWS account that is automatically a member of the organization whose credentials made the request. AWS Organizations automatically copies the information from the management account to the new member account
 */
export abstract class AccountBase extends Construct implements IAccount, IPolicyAttachmentTarget {
  public abstract readonly accountId: string;
  public abstract readonly accountName: string;
  public abstract readonly email: string;

  identifier(): string {
    return this.accountId;
  }

  currentParentId(): string {
    const parent = new AwsCustomResource(this, "ListParentsCustomResource", {
      onCreate: {
        service: "Organizations",
        action: "listParents", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#listParents-property
        region: "us-east-1",
        physicalResourceId: PhysicalResourceId.fromResponse("Parents.0.Id"),
        parameters: {
          ChildId: this.accountId,
        },
      },
      onUpdate: {
        service: "Organizations",
        action: "listParents", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#listParents-property
        region: "us-east-1",
        physicalResourceId: PhysicalResourceId.fromResponse("Parents.0.Id"),
        parameters: {
          ChildId: this.accountId,
        },
      },
      onDelete: {
        service: "Organizations",
        action: "listParents", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#listParents-property
        region: "us-east-1",
        physicalResourceId: PhysicalResourceId.fromResponse("Parents.0.Id"),
        parameters: {
          ChildId: this.accountId,
        },
      },
      installLatestAwsSdk: false,
      policy: AwsCustomResourcePolicy.fromSdkCalls({
        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
      }),
    });

    return parent.getResponseField("Parents.0.Id");
  }

  move(destinationParentId: string, sourceParentId: string): void {
    if (destinationParentId == sourceParentId) {
      return;
    }

    new AwsCustomResource(this, "MoveAccountCustomResource", {
      onUpdate: {
        service: "Organizations",
        action: "moveAccount", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#moveAccount-property
        region: "us-east-1",
        physicalResourceId: PhysicalResourceId.of(this.accountId),
        parameters: {
          AccountId: this.accountId,
          DestinationParentId: destinationParentId,
          SourceParentId: sourceParentId,
        },
      },
      installLatestAwsSdk: false,
      policy: AwsCustomResourcePolicy.fromSdkCalls({
        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
      }),
    });
  }
}

export interface AccountAttributes {
  readonly accountId: string;
  readonly parent: IParent;
}

export class Account extends AccountBase {
  /**
   * Import an existing account from account id.
   */
  public static fromAccountId(scope: Construct, id: string, attrs: AccountAttributes): IAccount {
    class Import extends AccountBase {
      public readonly accountId = attrs.accountId;
      public readonly accountName: string;
      public readonly email: string;

      public constructor() {
        super(scope, id);

        const account = new AwsCustomResource(this, "MoveAccountCustomResource", {
          onCreate: {
            service: "Organizations",
            action: "describeAccount", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#describeAccount-property
            region: "us-east-1",
            physicalResourceId: PhysicalResourceId.of(this.accountId),
            parameters: {
              AccountId: this.accountId,
            },
          },
          installLatestAwsSdk: false,
          policy: AwsCustomResourcePolicy.fromSdkCalls({
            resources: AwsCustomResourcePolicy.ANY_RESOURCE,
          }),
        });

        this.accountName = account.getResponseField("Account.Name");
        this.email = account.getResponseField("Account.Email");

        if (attrs.parent) {
          const sourceParentId = this.currentParentId();
          const targetParentId = attrs.parent.identifier();
          this.move(targetParentId, sourceParentId);
        }
      }
    }

    return new Import();
  }

  public readonly accountId: string;
  public readonly accountName: string;
  public readonly email: string;

  public constructor(scope: Construct, id: string, props: AccountProps) {
    super(scope, id);

    const { email, accountName, roleName, iamUserAccessToBilling, parent } = props;

    const accountProvider = AccountProvider.getOrCreate(this);
    const account = new CustomResource(this, `AccountProvider-${accountName}`, {
      serviceToken: accountProvider.provider.serviceToken,
      resourceType: "Custom::Organization_Account",
      properties: {
        Email: email,
        AccountName: accountName,
        RoleName: roleName || "OrganizationAccountAccessRole",
        IamUserAccessToBilling: iamUserAccessToBilling || IamUserAccessToBilling.ALLOW,
      },
    });
    this.accountId = account.getAtt("AccountId").toString();
    this.accountName = account.getAtt("AccountName").toString();
    this.email = email;

    if (parent) {
      const sourceParentId = this.currentParentId();
      const targetParentId = parent.identifier();
      this.move(targetParentId, sourceParentId);
    }
  }
}
