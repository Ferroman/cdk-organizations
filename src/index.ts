export * from "./account";
export * from "./organization";
export * from "./organizational-unit";
