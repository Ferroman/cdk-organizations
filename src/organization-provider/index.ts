export * from "./on-event-handler-function";
export * from "./organization-provider";
