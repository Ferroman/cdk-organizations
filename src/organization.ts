import { Construct } from "constructs";

/**
 * @see https://docs.aws.amazon.com/organizations/latest/userguide/orgs_getting-started_concepts.html#feature-set
 */
export enum FeatureSet {
  CONSOLIDATED_BILLING = "CONSOLIDATED_BILLING",
  ALL = "ALL",
}

export interface OrganizationProps {
  /**
   * Specifies the feature set supported by the new organization. Each feature set supports different levels of functionality.
   */
  featureSet?: FeatureSet;
}

/**
 * Creates an organization to consolidate your AWS accounts so that you can administer them as a single unit. An organization has one management account along with zero or more member accounts. You can organize the accounts in a hierarchical, tree-like structure with a root at the top and organizational units nested under the root. Each account can be directly in the root, or placed in one of the OUs in the hierarchy. An organization has the functionality that is determined by the feature set that you enable.
 *
 * @see https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_org_create.html#create-org
 */
export class Organization extends Construct {
  public constructor(scope: Construct, id: string, props: OrganizationProps) {
    super(scope, id);

    props;
  }
}
