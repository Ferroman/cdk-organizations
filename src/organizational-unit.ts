import { Construct } from "@aws-cdk/core";
import {
  AwsCustomResource,
  AwsCustomResourcePolicy,
  PhysicalResourceId,
  PhysicalResourceIdReference,
} from "@aws-cdk/custom-resources";

export interface OrganizationalUnitProps {
  /**
   * The friendly name to assign to the new OU.
   */
  readonly organizationalUnitName: string;

  /**
   * The unique identifier (ID) of the parent root or OU that you want to create the new OU in.
   */
  readonly parentId: string;

  /**
   * A list of tags that you want to attach to the newly created organizational unit. For each tag in the list, you must specify both a tag key and a value. You can set the value to an empty string, but you can't set it to null.
   */
  readonly tags: { [key: string]: string };
}

/**
 * A container for accounts within a root. An OU also can contain other OUs, enabling you to create a hierarchy that resembles an upside-down tree, with a root at the top and branches of OUs that reach down, ending in accounts that are the leaves of the tree. When you attach a policy to one of the nodes in the hierarchy, it flows down and affects all the branches (OUs) and leaves (accounts) beneath it. An OU can have exactly one parent, and currently each account can be a member of exactly one OU.
 *
 * <strong>You must first move all accounts out of the OU and any child OUs, and then you can delete the child OUs.</strong>
 */
export class OrganizationalUnit extends Construct {
  public constructor(scope: Construct, id: string, props: OrganizationalUnitProps) {
    super(scope, id);

    const { organizationalUnitName, parentId } = props;

    const tags: { Key: string; Value: string }[] = [];
    for (const [key, value] of Object.entries(props.tags)) {
      tags.push({
        Key: key,
        Value: value,
      });
    }

    new AwsCustomResource(this, "OrganizationalUnitCustomResource", {
      resourceType: "Custom::Organization_OrganizationalUnit",
      onCreate: {
        service: "Organization",
        action: "createOrganizationalUnit", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#createOrganizationalUnit-property
        region: "us-east-1",
        parameters: {
          Name: organizationalUnitName,
          ParentId: parentId,
          Tags: tags,
        },
        physicalResourceId: PhysicalResourceId.fromResponse("OrganizationalUnit.Id"),
      },
      onUpdate: {
        service: "Organization",
        action: "updateOrganizationalUnit", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#updateOrganizationalUnit-property
        region: "us-east-1",
        parameters: {
          Name: organizationalUnitName,
          OrganizationalUnitId: new PhysicalResourceIdReference(),
        },
        physicalResourceId: PhysicalResourceId.fromResponse("OrganizationalUnit.Id"),
      },
      onDelete: {
        service: "Organization",
        action: "deleteOrganizationalUnit", // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Organizations.html#deleteOrganizationalUnit-property
        region: "us-east-1",
        parameters: {
          OrganizationalUnitId: new PhysicalResourceIdReference(),
        },
      },
      policy: AwsCustomResourcePolicy.fromSdkCalls({ resources: AwsCustomResourcePolicy.ANY_RESOURCE }),
    });

    // TODO: https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_core.Tags.html
  }
}
