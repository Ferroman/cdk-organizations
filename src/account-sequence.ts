import { IAspect, Stack } from "aws-cdk-lib";
import { IConstruct } from "constructs";
import { Account } from "./account";

export class AccountSequence implements IAspect {
  private sequences: { [stackName: string]: Account[] } = {};

  visit(current: IConstruct): void {
    if (!(current instanceof Account)) {
      return;
    }

    const stackName = Stack.of(current).stackName;
    const sequence = this.sequences[stackName];

    if (sequence) {
      const previous = sequence[sequence.length - 1];
      current.node.addDependency(previous);
      sequence.push(current);
    }

    this.sequences[stackName] = [current];
  }
}
