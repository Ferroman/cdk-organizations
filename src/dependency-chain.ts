import { IAspect, Stack } from "aws-cdk-lib";
import { IConstruct } from "constructs";
import { Account } from "./account";
import { DelegatedAdministrator } from "./delegated-administrator";
import { EnableAwsServiceAccess } from "./enable-aws-service-access";
import { EnablePolicyType } from "./enable-policy-type";
import { OrganizationalUnit } from "./organizational-unit";
import { PolicyAttachment } from "./policy-attachment";

/**
 * Aspect to create dependency chain of organization resource that needs to be deployed sequentially
 * @experimental
 */
export class DependencyChain implements IAspect {
  private previous: { [stackName: string]: IConstruct } = {};

  visit(current: IConstruct): void {
    if (!this.needsChaining(current)) {
      return;
    }

    const stackName = Stack.of(current).stackName;

    if (this.previous[stackName]) {
      current.node.addDependency(this.previous[stackName]);
    }

    this.previous[stackName] = current;
  }

  private needsChaining(current: IConstruct): boolean {
    switch (true) {
      case current instanceof EnablePolicyType:
        return true;
      case current instanceof EnableAwsServiceAccess:
        return true;
      case current instanceof DelegatedAdministrator:
        return true;
      case current instanceof Account:
        return true;
      case current instanceof OrganizationalUnit:
        return true;
      case current instanceof PolicyAttachment:
        return true;
      default:
        return false;
    }
  }
}
