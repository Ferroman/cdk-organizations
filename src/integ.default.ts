import { App, Stack, Tags } from "aws-cdk-lib";
import { Account, IamUserAccessToBilling } from "./account";
import { FeatureSet, Organization } from "./organization";
import { OrganizationalUnit } from "./organizational-unit";
import { Policy, PolicyType } from "./policy";

const app = new App();
const stack = new Stack(app);

// Create an organization
const organization = new Organization(stack, "Organization", {
  featureSet: FeatureSet.ALL, // It's recommended to enable all features. It's required for service control policies (SCP)
});
// Enable AWS Service Access (requires FeatureSet: ALL)
organization.enableAwsServiceAccess("service-abbreviation.amazonaws.com");
organization.enableAwsServiceAccess("ssm.amazonaws.com");
organization.enableAwsServiceAccess("config-multiaccountsetup.amazonaws.com");

// Import an existing account
const account = new Account(stack, "ImportedAccount", {
  accountName: "test",
  email: "info+integ-test@pepperize.com",
  parent: organization.root,
});
// Enable a delegated admin account
account.delegateAdministrator("service-abbreviation.amazonaws.com");
account.delegateAdministrator("stacksets.cloudformation.amazonaws.com");
account.delegateAdministrator("config.amazonaws.com");

const projects = new OrganizationalUnit(stack, "ProjectsOU", {
  organizationalUnitName: "Projects",
  parent: organization.root,
});
new Account(stack, "Project1Account", {
  accountName: "SharedAccount",
  email: "info+project1@pepperize.com",
  iamUserAccessToBilling: IamUserAccessToBilling.DENY,
  parent: projects,
});

const project2 = new OrganizationalUnit(stack, "Project2OU", {
  organizationalUnitName: "Project2",
  parent: projects,
});
new Account(stack, "Project2DevAccount", {
  accountName: "Project 2 Dev",
  email: "info+project2-dev@pepperize.com",
  parent: project2,
});

// Enable the service control policy (SCP) type within the organization
organization.enablePolicyType(PolicyType.SERVICE_CONTROL_POLICY);
// Create and attach Service Control Policy (SCP)
const s3Policy = new Policy(stack, "S3Policy", {
  content: '{\n"Version":"2012-10-17","Statement":{\n"Effect":"Allow","Action":"s3:*"\n}\n}',
  description: "Enables admins of attached accounts to delegate all S3 permissions",
  policyName: "AllowAllS3Actions",
  policyType: PolicyType.SERVICE_CONTROL_POLICY,
});
organization.attachPolicy(s3Policy);

// Enable the tag policy type within the organization
organization.enablePolicyType(PolicyType.TAG_POLICY);
// Create and attach tag Policy
const tagPolicy = new Policy(stack, "TagPolicy", {
  content: '{\n"tags":{\n"CostCenter":{\n"tag_key":{\n"@@assign":"CostCenter"\n}\n}\n}\n}',
  description: "Defines the CostCenter tag key",
  policyName: "CostCenterTag",
  policyType: PolicyType.TAG_POLICY,
});
// Attach policy to an organizational unit (OU)
projects.attachPolicy(tagPolicy);
// Attach policies to an account
account.attachPolicy(tagPolicy);
account.attachPolicy(s3Policy);

// Tagging AWS organization resources of this stack
Tags.of(stack).add("tagKey", "tagValue");

export { app, stack };
