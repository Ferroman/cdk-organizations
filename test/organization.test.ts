import { Template } from "@aws-cdk/assertions";
import { App, Aspects, Stack } from "@aws-cdk/core";
import { AwsSolutionsChecks } from "cdk-nag";
import { FeatureSet, Organization } from "../src";

describe("Organization", () => {
  it("Should match snapshot", () => {
    // Given
    const app = new App();
    const stack = new Stack(app, "Stack");

    // When
    new Organization(stack, "Organization", {
      featureSet: FeatureSet.ALL,
    });

    // Then
    const template = Template.fromStack(stack);
    expect(template).toMatchSnapshot();
  });

  it("Should comply to best practices", () => {
    // Given
    const app = new App();
    const stack = new Stack(app, "Stack");

    // When
    new Organization(stack, "Organization", {});

    // Then
    Aspects.of(app).add(new AwsSolutionsChecks({ verbose: true }));
  });
});
