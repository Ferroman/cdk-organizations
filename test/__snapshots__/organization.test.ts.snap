// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization Should match snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8dArtifactHash6BDDF757": Object {
      "Description": "Artifact hash for asset \\"70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8d\\"",
      "Type": "String",
    },
    "AssetParameters70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8dS3BucketF0537962": Object {
      "Description": "S3 bucket for asset \\"70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8d\\"",
      "Type": "String",
    },
    "AssetParameters70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8dS3VersionKey46386A78": Object {
      "Description": "S3 key for asset version \\"70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8d\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": Object {
      "DependsOn": Array [
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8dS3BucketF0537962",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8dS3VersionKey46386A78",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters70893b631249dc61260989e92e90d60ae94fbbec490a1e065680d77383084d8dS3VersionKey46386A78",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "OrganizationOrgCustomResourceCustomResourcePolicy17A1A1A4": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "organization:CreateOrganization",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "OrganizationOrgCustomResourceCustomResourcePolicy17A1A1A4",
        "Roles": Array [
          Object {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "OrganizationOrgCustomResourceFE5C487D": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "OrganizationOrgCustomResourceCustomResourcePolicy17A1A1A4",
      ],
      "Properties": Object {
        "Create": "{\\"service\\":\\"Organization\\",\\"action\\":\\"createOrganization\\",\\"region\\":\\"us-east-1\\",\\"physicalResourceId\\":{\\"responsePath\\":\\"Organization.Id\\"}}",
        "InstallLatestAwsSdk": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
      },
      "Type": "Custom::AWS",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
